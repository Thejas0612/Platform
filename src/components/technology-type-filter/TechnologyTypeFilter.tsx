import React, {FunctionComponent, useState} from "react";
import {
    Button,
    Grid,
    MenuItem,
    Select, SelectChangeEvent,
    Typography
} from "@mui/material";
import { makeStyles } from "@mui/styles";
import TuneIcon from "@mui/icons-material/Tune";
import CloseIcon from "@mui/icons-material/Close";
import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";

interface Option{
    value: string;
    label: string;
}

export interface TechnologyTypeFilterProps {
    onClose?: (sortByValue: string) => void;
    sortOptions: Option[];
}
const useStyles = makeStyles({
    button: {
        textTransform: "none",
        color: "black",
        padding: "20px"
    }
});
export const TechnologyTypeFilter: FunctionComponent<TechnologyTypeFilterProps> = ({sortOptions, onClose = () => {}}) => {
    const [expanded, setExpanded] = useState<boolean>(false);
    const [sortValue, setSortValue] = useState("");
    const classes = useStyles();

    const handleFilterOpen = () => () => {
        setExpanded(!expanded);
        expanded && onClose(sortValue);
        };

    const handleSortChange = (event: SelectChangeEvent<unknown>) => {
        const selectedValue = event.target.value as string;
        setSortValue(selectedValue);
    };

    return (
        <Grid container justifyContent="flex-end" alignItems="center">
            <Grid item>
                {expanded ? (<Grid container direction="row" alignItems="center">
                    <Typography sx={{ paddingRight: "20px"}}> Sort By: </Typography>
                    <Select
                        IconComponent={KeyboardArrowDownIcon}
                        onChange={handleSortChange}
                        sx={{minWidth: "160px", fontSize: "12px"}}
                        value={sortValue}
                    >
                        {sortOptions.map((option, index) => (
                            <MenuItem key={index} value={option.value}>
                                {option.label}
                            </MenuItem>
                        ))}
                    </Select>
                </Grid>) : <></>}
            </Grid>
            <Grid item>
                <Button
                    variant="text"
                    endIcon={expanded ? <CloseIcon/> : <TuneIcon/>}
                    onClick={handleFilterOpen()}
                    className={classes.button}
                >
                    <Typography>Filters </Typography>
                </Button>
            </Grid>
        </Grid>
        );
};