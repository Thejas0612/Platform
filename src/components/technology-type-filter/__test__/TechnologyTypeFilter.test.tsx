import {TechnologyTypeFilter, TechnologyTypeFilterProps} from "../TechnologyTypeFilter";
import {fireEvent, render} from "@testing-library/react";

const PROPS: TechnologyTypeFilterProps = {
    sortOptions: [{ label: "Recommended", value: "recommended"}]
}

describe("<TechnologyTypeFilter />", () => {
    it("should render the filter with default state", () => {
        const {container} = render(<TechnologyTypeFilter {...PROPS} />)

        expect(container).toMatchSnapshot()
    });

    it("should render the sort by filter after click the filter", () => {
        const { getByText, getByRole } = render(<TechnologyTypeFilter {...PROPS} />);
        const button = getByText("Filters");
        fireEvent.click(button);
        const select = getByRole("combobox");
        expect(select).toBeInTheDocument();
    });

    it("should fire onClose event when clicked.", async () => {
        const onCloseMock = jest.fn()
        const { getByText } = render(<TechnologyTypeFilter onClose={onCloseMock} {...PROPS}/>);
        const button = getByText("Filters");
        fireEvent.click(button);
        fireEvent.click(button);
        expect(onCloseMock).toHaveBeenCalled();
    })
})