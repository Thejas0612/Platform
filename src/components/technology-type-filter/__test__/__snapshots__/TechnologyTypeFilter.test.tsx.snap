// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TechnologyTypeFilter /> should render the filter with default state 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-1bizp5f-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium makeStyles-button-1 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Filters 
        </p>
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="TuneIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17v2h6v-2zM3 5v2h10V5zm10 16v-2h8v-2h-8v-2h-2v6zM7 9v2H3v2h4v2h2V9zm14 4v-2H11v2zm-6-4h2V7h4V5h-4V3h-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
